chain/chain/src/chain.rs

```diff
@@ -782,8 +782,17 @@ impl Chain {
         } else {
             debug!(target: "chain", "Downloading state for {:?}, I'm {:?}", shards_to_state_sync, me);
 
+            let protocol_version =
+                self.epoch_manager.get_epoch_protocol_version(block.header().epoch_id())?;
+            let sync_hash = if checked_feature!("stable", StateSyncHashUpdate, protocol_version) {
+                None
+            } else {
+                Some(*block.header().hash())
+            };
             let state_sync_info = StateSyncInfo {
-                epoch_tail_hash: *block.header().hash(),
+                state_sync_version: 0,
+                epoch_first_block: *block.header().hash(),
```

Comment:
It's non obvious that the block is guaranteed to be the first in this epoch - can you clarify that somehow?

Reply:
Agreed, p non obvious. This function is only called when the block is the first of an epoch, and this implicit assumption here that it’s the first of an epoch existed even before this PR. Renamed the function argument and added a comment